File: Wk5Scala/src/abstractfactory/ParserFactoryProducer.scala
package abstractfactory

import com.sun.deploy.xml.XMLParser

/**
  * Created by geoff_000 on 06/02/2016.
  */
object ParserFactoryProducer extends AbstractParserFactory {
  override def getParserInstance(parserType: String): XMLParser = {
    if (parserType.equalsIgnoreCase("NYCFactory")){

    }

  }
}

File: Wk5Java/src/abstractfactoryjava/ParserFactoryProducer.java
package abstractfactoryjava;

/**
 * Created by geoff_000 on 06/02/2016.
 */
public class ParserFactoryProducer {

    public static AbstractParserFactory getFactory(String type) {
        AbstractParserFactory fac = null;
        if (type.equalsIgnoreCase("NYCFactory")) {
            fac = new NYCParserFactory();
        } else {
            fac = new LondonParserFactory();
        }
        return fac;

    }
}
